java/messages/src/main/java/com/nedap/retail/messages/epcis/v1_1/EpcisEventContainer.java

```diff
@@ -17,12 +18,11 @@
     public EpcisEventContainer() {
     }
 
-    public EpcisEventContainer(final List<EpcisEvent> events) {
-        this.events = events;
+    public EpcisEventContainer(final List<? extends EpcisEvent> events) {
+        this.events = new ArrayList<>(events);
     }
 
     public EpcisEventContainer(final EpcisEvent epcisEvent) {
-        events = new ArrayList<>();
-        events.add(epcisEvent);
+        events = new ArrayList<>(Arrays.asList(epcisEvent));
```

Comment:
Arrays.asList() already returns a new ArrayList, there is no need to create another one.